File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.py
from sortedcontainers import SortedList

class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        mp = {}
        ans = []

        for i in range(0, len(A)):
            if A[i] not in mp:
                mp[A[i]] = 0
            mp[A[i]] += 1
            if B[i] not in mp:
                mp[B[i]] = 0
            mp[B[i]] -= 1

            cnt = 0
            for aa, bb in mp.items():
                cnt += abs(bb)
            
            ans.append(int((i + 1) - (cnt / 2)))

        return ans

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        vector<int> ans(A.size());

        set<int> set;
        map<int, int> mp;
        for(int i = 0; i < A.size(); i++) {
            mp[A[i]]++;
            if(mp[A[i]] == 1) set.insert(A[i]);
            if(mp[A[i]] == 0) set.erase(A[i]);
            mp[B[i]]--;
            if(mp[B[i]] == 0) set.erase(B[i]);
            if(mp[B[i]] == -1) set.insert(B[i]);

            ans[i] = (i + 1) - set.size() / 2;
        }

        return ans;
    }
};